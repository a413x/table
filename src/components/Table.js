import React, { Component } from "react";
import PropTypes from 'prop-types';

import TableHead from './TableHead.js';

class Table extends Component{
  onRowClick(selectedUser){
    this.props.onRowClick(selectedUser);
  }
  render(){
    const {data,columns} = this.props;
    return(
      <table className = 'table table-hover'>
        <TableHead columns = {columns}
          sortFunc = {this.props.sortFunc}/>
        <tbody>
          {data.map((dataObj,ind)=>
            <tr key={ind} onClick = {()=>this.onRowClick(dataObj)}>
              {columns.map((colObj,ind) =>
                <td key={ind}>{dataObj[colObj.field]}</td>
              )}
            </tr>
          )}
        </tbody>
      </table>
    );
  }
}

Table.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
  sortFunc: PropTypes.func.isRequired,
  onRowClick: PropTypes.func.isRequired
};

export default Table;
