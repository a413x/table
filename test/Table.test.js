import React from "react";
import {shallow,mount} from 'enzyme';

import Table from "../src/components/Table";
import data from './data.js';

describe('tests for Table component',()=>{
  const tableColumns = data.tableColumns;
  const fakeData = data.fakeData;
  const mockOnRowClick = jest.fn();

  const props = {
    data: fakeData,
    columns: tableColumns,
    sortFunc: () => {},
    onRowClick: mockOnRowClick
  };

  const tableContainer = shallow(<Table {...props}/>);

  it('renders properly',()=>{
    expect(tableContainer).toMatchSnapshot();
  });

  it('onRowClick called when clicked on a row',()=>{
    tableContainer.find('tr').first().simulate('click');
    expect(mockOnRowClick).toHaveBeenCalledTimes(1);
  });
});
