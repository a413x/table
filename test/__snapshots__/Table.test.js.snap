// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Table component renders properly 1`] = `
<table
  className="table table-hover"
>
  <TableHead
    columns={
      Array [
        Object {
          "field": "id",
          "title": "#",
          "type": "number",
        },
        Object {
          "field": "firstName",
          "title": "First name",
          "type": "text",
        },
        Object {
          "field": "lastName",
          "title": "Last name",
          "type": "text",
        },
        Object {
          "field": "email",
          "title": "E-mail",
          "type": "email",
        },
        Object {
          "field": "phone",
          "title": "Phone",
          "type": "phone",
        },
      ]
    }
    sortFunc={[Function]}
  />
  <tbody>
    <tr
      key="0"
      onClick={[Function]}
    >
      <td
        key="0"
      >
        1
      </td>
      <td
        key="1"
      >
        A
      </td>
      <td
        key="2"
      >
        B
      </td>
      <td
        key="3"
      >
        a@aa.aa
      </td>
      <td
        key="4"
      >
        11111111111
      </td>
    </tr>
    <tr
      key="1"
      onClick={[Function]}
    >
      <td
        key="0"
      >
        2
      </td>
      <td
        key="1"
      >
        C
      </td>
      <td
        key="2"
      >
        D
      </td>
      <td
        key="3"
      >
        b@bb.bb
      </td>
      <td
        key="4"
      >
        21111111111
      </td>
    </tr>
    <tr
      key="2"
      onClick={[Function]}
    >
      <td
        key="0"
      >
        3
      </td>
      <td
        key="1"
      >
        E
      </td>
      <td
        key="2"
      >
        F
      </td>
      <td
        key="3"
      >
        c@cc.cc
      </td>
      <td
        key="4"
      >
        31111111111
      </td>
    </tr>
    <tr
      key="3"
      onClick={[Function]}
    >
      <td
        key="0"
      >
        4
      </td>
      <td
        key="1"
      >
        G
      </td>
      <td
        key="2"
      >
        H
      </td>
      <td
        key="3"
      >
        d@dd.dd
      </td>
      <td
        key="4"
      >
        41111111111
      </td>
    </tr>
    <tr
      key="4"
      onClick={[Function]}
    >
      <td
        key="0"
      >
        5
      </td>
      <td
        key="1"
      >
        I
      </td>
      <td
        key="2"
      >
        J
      </td>
      <td
        key="3"
      >
        e@ee.ee
      </td>
      <td
        key="4"
      >
        51111111111
      </td>
    </tr>
    <tr
      key="5"
      onClick={[Function]}
    >
      <td
        key="0"
      >
        6
      </td>
      <td
        key="1"
      >
        K
      </td>
      <td
        key="2"
      >
        L
      </td>
      <td
        key="3"
      >
        f@ff.ff
      </td>
      <td
        key="4"
      >
        61111111111
      </td>
    </tr>
  </tbody>
</table>
`;
